refresh_compile_commands.bzl

```diff
@@ -23,61 +23,81 @@ refresh_compile_commands(
         # Or a dict of targets and any flags required to build:
             # (No need to add flags already in .bazelrc. They're automatically picked up.)
             # targets = {
-            #   "//:my_output_1": "--important_flag1 --important_flag2=true", 
+            #   "//:my_output_1": "--important_flag1 --important_flag2=true",
             #   "//:my_output_2": "",
             # },
         # If you don't specify a target, that's fine (if it works for you); compile_commands.json will default to containing commands used in building all possible targets. But in that case, just bazel run @hedron_compile_commands//:refresh_all
         # Wildcard target patterns (..., *, :all) patterns *are* allowed, like in bazel build
             # For more, see https://docs.bazel.build/versions/main/guide.html#specifying-targets-to-build
+
+    # Optional attributes :
+
+    # To omit entries for sources from external workspaces (dependencies)
+        # This defaults to False
+    exclude_external_sources = False
+
+    # To omit entries for some or all headers
+        # Defaults to None (always include all headers)
+        # Some tools such as ccls work better without header entries, whereas others such as clangd require these (see https://github.com/clangd/clangd/issues/123 for why)
+        # all - will omit entires for any and all header files
+        # system - will omit system headers (/usr/include, etc.), but keep entries for headers in your workspace, and from any external workspaces (dependencies)
+        # external - will omit any external headers, but keep system and your workspace's
+        # external_and_system - will keep just your workspace's headers
+        #
+        # Note that any generated files from your workspace in bazel-out, etc. will still be included - these are considered part of your workspace's headers
+    exclude_external_workspaces = "all" | "external" | "system" | "external_and_system"
```

Comment:
Could I ask you give this another round of polish?
(e.g. might want to say *why* you might want each option where that's missing, comment out optional entries, so people can just copy and uncomment to toggle, etc.) 

Do you think we should add anything to the README to aid discoverability?
(Perhaps like a "Using CCLS or another tool that doesn't need compile commands entries for headers?" as in our discussion? I can also do this later.)

Reply:
Updated.

Can I ask that you make any further refinements here if needed, or we update the docs as a follow-up PR?

I don't have much more time to devote to this PR, so would like to see it land soon.